lib/rspec/support/object_inspector.rb

```diff
@@ -13,6 +13,8 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
```

Comment:
Generally we would write this as `RSpec::Mocks::Double === object` which prevents the target tampering with the `==` operator on it's class (which is a possibility)